packages/keyring-api/src/eth/v2/eth-keyring-wrapper.ts

```diff
@@ -0,0 +1,213 @@
+        const [data] = params;
+        return this.inner.signPersonalMessage(hexAddress, data as Hex);
+      }
+
+      case `${EthMethod.SignTypedDataV1}`: {
+        if (!this.inner.signTypedData) {
+          throw new Error('Keyring does not support signTypedData');
+        }
+        assert(params, EthSignTypedDataV1ParamsStruct);
+        const [, data] = params;
+        return this.inner.signTypedData(hexAddress, data, {
+          version: SignTypedDataVersion.V1,
+        });
+      }
+
+      case `${EthMethod.SignTypedDataV3}`: {
+        if (!this.inner.signTypedData) {
+          throw new Error('Keyring does not support signTypedData');
+        }
+        assert(params, EthSignTypedDataParamsStruct);
+        const [, data] = params;
+        return this.inner.signTypedData(
+          hexAddress,
+          // TODO: Improve typing to ensure data matches MessageTypes
+          data as TypedMessage<MessageTypes>,
+          {
+            version: SignTypedDataVersion.V3,
+          },
+        );
+      }
+
+      case `${EthMethod.SignTypedDataV4}`: {
+        if (!this.inner.signTypedData) {
+          throw new Error('Keyring does not support signTypedData');
+        }
+        assert(params, EthSignTypedDataParamsStruct);
+        const [, data] = params;
+        return this.inner.signTypedData(
+          hexAddress,
+          // TODO: Improve typing to ensure data matches MessageTypes
+          data as TypedMessage<MessageTypes>,
+          {
+            version: SignTypedDataVersion.V4,
+          },
+        );
+      }
+
+      case `${EthKeyringMethod.Decrypt}`: {
+        if (!this.inner.decryptMessage) {
+          throw new Error('Keyring does not support decryptMessage');
+        }
+        assert(params, EthDecryptParamsStruct);
+        const [encryptedData] = params;
+        return this.inner.decryptMessage(hexAddress, encryptedData);
+      }
+
+      case `${EthKeyringMethod.GetEncryptionPublicKey}`: {
+        if (!this.inner.getEncryptionPublicKey) {
+          throw new Error('Keyring does not support getEncryptionPublicKey');
+        }
+        return this.inner.getEncryptionPublicKey(hexAddress);
+      }
+
+      case `${EthKeyringMethod.GetAppKeyAddress}`: {
+        if (!this.inner.getAppKeyAddress) {
+          throw new Error('Keyring does not support getAppKeyAddress');
+        }
+        assert(params, EthGetAppKeyAddressParamsStruct);
+        const [origin] = params;
+        return this.inner.getAppKeyAddress(hexAddress, origin);
+      }
+
+      case `${EthKeyringMethod.SignEip7702Authorization}`: {
+        if (!this.inner.signEip7702Authorization) {
+          throw new Error('Keyring does not support signEip7702Authorization');
+        }
+        assert(params, EthSignEip7702AuthorizationParamsStruct);
+        const [authorization] = params;
+        return this.inner.signEip7702Authorization(
+          hexAddress,
+          authorization as [number, Hex, number],
```

Comment:
I thought we already changed the struct to only accept `number` and not `string`? We should not need any cast here, or that means our tests are invalid most probably?